### PR TITLE
Modernise

### PR DESCRIPTION
**Pull request by Rich-Harris**
- switch to pnpm
- switch to ESM
- get rid of the build step (except `tsc`)
- get rid of mocha in favour of `node:test`
- add prettier
- add github CI
- add `pkg.exports`

there's way more that can be done (i haven't changed any of the API, even though it's arguably pretty wacky by modern standards) but i don't want to sink too much time into this

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+on:
+  push:
+    branches: [master]
+  pull_request:
+permissions:
+  contents: read # to fetch code (actions/checkout)
+
+env:
+  # We only install Chromium manually
+  PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: '1'
```
Comment (benmccann) [inline .github/workflows/ci.yml:11]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            os: ubuntu-latest
+          - node-version: 20
+            os: ubuntu-latest
+          - node-version: 22
+            os: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: pnpm/action-setup@v4
+      - uses: actions/setup-node@v3
```
Comment (benmccann) [inline .github/workflows/ci.yml:34]:
```suggestion
      - uses: actions/checkout@v4
      - uses: pnpm/action-setup@v4
      - uses: actions/setup-node@v4
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      - run: pnpm test
+        env:
+          CI: true
+  Lint:
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v3
+      - uses: pnpm/action-setup@v4
+      - uses: actions/setup-node@v3
```
Comment (benmccann) [inline .github/workflows/ci.yml:48]:
```suggestion
      - uses: actions/checkout@v4
      - uses: pnpm/action-setup@v4
      - uses: actions/setup-node@v4
```

**Comment by Rich-Harris** [comment]
got some windows failures but i'm not going to look into them now, i'm done for the day

**Comment by benmccann** [comment]
Just FYI, Simon managed to rip sorcery out of `svelte-preprocess`, so you don't necessarily need to bang your head on this too long if it ends up being troublesome

**Comment by Rich-Harris** [comment]
I know, but I've come this far...

**Pull request merged**